src/optimizer/stats/table_stats_collector.cpp

```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// table_stats_collector.cpp
+//
+// Identification: src/optimizer/stats/table_stats_collector.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "optimizer/stats/table_stats_collector.h"
+
+#include <memory>
+
+#include "common/macros.h"
+#include "storage/tile_group.h"
+#include "storage/tile_group_header.h"
+#include "type/types.h"
+#include "type/value.h"
+
+namespace peloton {
+namespace optimizer {
+
+TableStatsCollector::TableStatsCollector(storage::DataTable* table)
+    : table_(table),
+      column_stats_collectors_{},
+      active_tuple_count_{0},
+      column_count_{0} {}
+
+TableStatsCollector::~TableStatsCollector() {}
+
+void TableStatsCollector::CollectColumnStats() {
+  schema_ = table_->GetSchema();
+  column_count_ = schema_->GetColumnCount();
+
+  // Ignore empty table
+  if (column_count_ == 0) {
+    return;
+  }
+
+  InitColumnStatsCollectors();
+
+  size_t tile_group_count = table_->GetTileGroupCount();
+  // Collect stats for all tile groups.
+  for (size_t offset = 0; offset < tile_group_count; offset++) {
+    std::shared_ptr<storage::TileGroup> tile_group =
+        table_->GetTileGroup(offset);
+    storage::TileGroupHeader* tile_group_header = tile_group->GetHeader();
+    oid_t tuple_count = tile_group->GetAllocatedTupleCount();
+    active_tuple_count_ += tile_group_header->GetActiveTupleCount();
```

Comment:
Each table has multiple tile groups. We get the active tuple count of each tile group, and add them together to get the total number of active tuples as the `active_tuple_count_`.